azurerm/internal/services/datafactory/data_factory_resource.go

```diff
@@ -410,6 +453,27 @@ func expandDataFactoryRepoConfiguration(d *pluginsdk.ResourceData) (bool, datafa
 	return false, nil
 }
 
+func expandDataFactoryGlobalParameters(input []interface{}) (map[string]*datafactory.GlobalParameterSpecification, error) {
+	if len(input) == 0 || input[0] == nil {
```

Comment:
We can remove this whole check because there can be any number of `global_parameters` specified. The for loop should handle any potential nil checks on individual items